ios/EmailShare.m

```diff
@@ -32,23 +63,109 @@ - (void)shareSingle:(NSDictionary *)options
+                        // file name should not include extension
+                        if(URL.isFileURL){
+                            NSString *ext = [[URL.absoluteString componentsSeparatedByString:@"."] lastObject];
+                            
+                            filename = [filename stringByAppendingString: [@"." stringByAppendingString:ext]];
+                        }
+                        else if (isDataScheme){
+                            NSString *ext = [self getExtensionFromBase64: URL.absoluteString];
+                            
+                            if(ext){
+                                filename = [filename stringByAppendingString: [@"." stringByAppendingString:ext]];
+                            }
+                        }
+                    }
+                    else if(URL.fileURL){
+                        NSArray *parts = [URL.absoluteString componentsSeparatedByString:@"/"];
+                        filename = [parts lastObject];
+                    }
+                    
+                    [mc addAttachmentData:data mimeType:mime fileName:filename];
 
-        if ([[UIApplication sharedApplication] canOpenURL: whatsappURL]) {
-            [[UIApplication sharedApplication] openURL: whatsappURL];
-            successCallback(@[]);
-        } else {
-            // Cannot open email
-            NSLog(@"Cannot open email");
+                } else {
+                    // if not a file, just append it to message
+                    message = [message stringByAppendingString: [@" " stringByAppendingString: [RCTConvert NSString:options[@"url"]]] ];
+                }
+            }
+                   
         }
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            UIViewController *ctrl = RCTPresentedViewController();
+          
+            // Present mail view controller on screen
+            [ctrl presentViewController:mc animated:YES completion:NULL];
+            
+            // We could fire this either here or
+            // on the finish delegate.
+            // For now, call it here for consistency with
+            // GenericShare.shareSingle
+            successCallback(@[]);
+        });
     }
+}
 
+-(void)mailComposeController:(MFMailComposeViewController *)controller didFinishWithResult:(MFMailComposeResult)result error:(nullable NSError *)error
+{
+// TODO: Do something with this
+//    switch (result)
+//    {
+//        case MFMailComposeResultCancelled:
+//            NSLog(@"Mail cancelled");
+//            break;
+//        case MFMailComposeResultSaved:
+//            NSLog(@"Mail saved");
+//            break;
+//        case MFMailComposeResultSent:
+//            NSLog(@"Mail sent");
+//            break;
+//        case MFMailComposeResultFailed:
+//            NSLog(@"Mail sent failure: %@", [error localizedDescription]);
+//            break;
+//        default:
+//            break;
+//    }
```

Comment:
We can remove this function?

Reply:
We need the handler so the share/email view can be closed after it's used.  We can certainly remove the commented code though.  I was wondering if there could be any use for it, but then I realized that anything done here wouldn't really be compatible with the rest of the share alternatives nor its Android version.

For example, this could be used to trigger the success event only after the user has closed/sent the email, and we could pass an enum to indicate what did the user do, but this wouldn't be possible on Android...

Reply:
I've removed the comments.